### ISSUE TITLE
Option to open URL in Incognito / Private Mode

### ISSUE THREAD

**Issue by polvmsn** [keepassxreboot/keepassxc#9468]
Please make KeepassXC open web entries in Incognito / Private browsing as additional action in right click menu on entry ... 

See Keepass menu
![image](https://github.com/keepassxreboot/keepassxc/assets/134379642/6f16e083-4cf0-40d9-9992-cd70d2c43c94)

I know that entries  can be manipulated with --incognito or --private but is not practical for daily use.

**Comment by varjolintu** [comment]
FYI: Started working on this.

**Comment by varjolintu** [comment]
In Linux `xdg-open` does not accept any command line arguments (`-private-window` or `-incognito` are needed). If anyone has a way to get the default browser binary path using some library, any hints are welcome.

**Comment by michaelk83** [comment]
Putting together some Googling results:
1. `xdg-mime query default text/html` to get the `.desktop` file name.
2. Scan through `~/.local/share/` and `XDG_DATA_DIRS` to find the actual file under the `applications` subfolder in each of the possible locations. For example, `~/.local/share/applications/`, `/usr/local/share/applications/`, `/usr/share/applications/`. First one takes precedence. Note that `XDG_DATA_DIRS` specifies only the system data dirs precedence order, it doesn't include `~/.local/share/`, which takes precedence over the system folders.
3. Read the `Exec` line of the `.desktop` file.
4. Split away any standard parameters. For example, in my case the full line is `Exec=/opt/firefox/firefox-bin %u`, probably `%u` stands for "URI".

https://unix.stackexchange.com/a/251055
https://wiki.archlinux.org/title/desktop_entries#Application_entry
https://specifications.freedesktop.org/desktop-entry-spec/desktop-entry-spec-latest.html#file-naming

**Comment by varjolintu** [comment]
@michaelk83 Yes, that's the way I also figured out, although I'd prefer `xdg-settings get default-web-browser`. Scanning the directories and reading the `Exec` from files is a bit ugly. Plus in my environment the `Exec` only showed `firefox -new-window`. That means the binary path still needs to be identified via `which` or something similar. A simple API call would be pretty nice :) I alread implemented macOS side and that was pretty easy to do.

**Comment by michaelk83** [comment]
I saw `xdg-settings`, but didn't dig into it. `xdg-settings get default-web-browser` gives me the same thing as `xdg-mime query default text/html`. So you'd still need to scan the folders etc.

For C++ there is at least https://github.com/azubieta/xdg-utils-cxx

**Comment by michaelk83** [comment]
Btw, if the `Exec` only says `firefox`, that means it's in one of the `PATH` folders. So you can run it as is. You don't need to use `which` to get the absolute path.

**Comment by varjolintu** [comment]
> I saw `xdg-settings`, but didn't dig into it. `xdg-settings get default-web-browser` gives me the same thing as `xdg-mime query default text/html`. So you'd still need to scan the folders etc.
> 
> For C++ there is at least https://github.com/azubieta/xdg-utils-cxx

I also found the `xdg-utils-cxx`. Need to check if it has all the necessary things.

EDIT: Another option would be just to check the `.desktop` file and assume the binary name from that, and using it without any paths.

**Comment by varjolintu** [comment]
For Linux/Unix I solved this by just using `xdg-settings` from `QProcess` and getting the file location with `QStandardPaths::locate(QStandardPaths::ApplicationsLocation, applicationName);`. After that the content is parsed looking for `Exec=` line. Seems to work fine.

**Comment by chriszuercher** [comment]
What is the state on this? I would like to use this feature because it is one of the few missing pieces from Keepass which I used frequently (e.g. for banking sessions)

**Comment by varjolintu** [comment]
> What is the state on this? I would like to use this feature because it is one of the few missing pieces from Keepass which I used frequently (e.g. for banking sessions)

The PR has been set for the KeePassXC 2.8.0 milestone.
